ipn/ipnlocal/peerapi.go

```diff
@@ -1104,15 +1106,52 @@
 	return nil
 }
 
+// httpResponseWrapper wraps an http.ResponseWrite and
+// stores the status code and content length.
+type httpResponseWrapper struct {
+	http.ResponseWriter
+	statusCode    int
+	contentLength int64
+}
+
+// WriteHeader implements the WriteHeader interface.
+func (hrw *httpResponseWrapper) WriteHeader(status int) {
+	hrw.statusCode = status
+	hrw.ResponseWriter.WriteHeader(status)
+}
+
+// Write implements the Write interface.
+func (hrw *httpResponseWrapper) Write(b []byte) (int, error) {
+	n, err := hrw.ResponseWriter.Write(b)
```

Comment:
This flags some pre-existing code (like c2n `echo` handler), but also something in TailFS: https://github.com/tailscale/tailscale/blob/b06aca1dc0aed843252cd94054026b5ade8ac14a/tailfs/tailfsimpl/compositedav/propfind.go#L44-L44
I don't know much about webdav, but that code writes back something derived from the incoming request path back into the response.
Is there a way to craft the request path to inject something nasty in there, like `</D:href>evil.com`? Do we maybe need some escaping of the value, if it's not done already?

Reply:
Since this is pre-existing code, will follow this up in a separate PR.

Reply:
I'm not concerned, The path prefix that we're injecting came from the client to whom we're sending the response. Also, if the client were to use a bogus path, it would result in us returning a 404.